rts/Game/UI/SelectionKeyHandler.cpp

```diff
@@ -432,6 +432,26 @@ void CSelectionKeyHandler::DoSelection(std::string selectString)
 			return;
 		} break;
 
+		case hashString("SelectOneNearestMouse"): {
+			if (selection.empty())
+				return;
+
+			const float gndDist = CGround::LineGroundCol(camera->GetPos(), camera->GetPos() + mouse->dir * camera->GetFarPlaneDist(), false);
+			float3 mousePosition = camera->GetPos() + mouse->dir * gndDist;
+
+			CUnit* closest = nullptr;
+			float closestDistance = 0;
+			for (auto* unit : selection) {
+				float distance = mousePosition.SqDistance(unit->pos);
+				if (!closest || distance < closestDistance) {
+					closestDistance = distance;
+				closest = unit;
```

Comment:
> I would like to understand the context of this change. Like: what problem are we trying to solve with this? It makes it easier to review the change.

When talking to a friend, he said he wanted a key to select the closest constructor. Right now, the only way to do anything similar to this is with `FromMouse_500` or so. I thought this might be helpful. But if you think this is not a useful selection command, I'm very much OK with not merging it ;)

Reply:
> Indentation looks off here,

fixed.